ush/templates/FV3LAM_wflow.xml

```diff
@@ -3213,6 +3213,12 @@ MODULES_RUN_TASK_FP script.
     <envar><name>nens</name><value><cyclestr>0</cyclestr></value></envar>
 {%- endif %}
 
+{%- if do_retro %}
+    <dependency>
+       <taskdep task="&RUN_FCST_TN;_prod"/>
+    </dependency>
+{%- endif %}
+
```

Comment:
@HaidaoLin-NOAA Why do we need this? The clean script is to clean old data before the current cycle. I would think it should not depend on any tasks in the current cycle.

Reply:
@guoqing-noaa, In retro runs, if I use short clean window (e.g, 6h, clean files 6h ago compared to current cycle run) and a relative larger cycletjrpttle number (e.g, 12 or larger than 6), if no dependence, the clean script would delete files for those cycles that are not finished yet.  I had this experience.  If you use very long clean window and relative smaller cyclethrottle (clean window > cyclethrottle ,maybe this is more often), then you would be OK.  I think it would be safer to have a dependency for the clean task and it won't hurt anything.

Reply:
@guoqing-noaa 1) OK. 2) Yes, I have spin-up cycles in my 13km runs. What do you mean if using spin-up cycles, clean.sh does not fit your retros?  BTW: I did not use the original script for cleaning but using a modified version to fit my needs, but I still use the task in the workflow.  3) I am fine not to include the dependence for the clean task. :-)